### ISSUE TITLE
[REQUEST] Group Devices

### ISSUE THREAD

**Issue by artfer** [seriousm4x/UpSnap#97]
**The problem you currently have with this project**
When many devices are added, the page gets a bit cluttered and confusing. 

**Describe the solution you'd like**
Add "Group" text box to the parameters of the "Add New Device" menu and group them in the main page.

**Comment by gaby** [comment]
I just realized, I have the same problem as this ticket

**Comment by seriousm4x** [comment]
Thanks for suggesting this. I've added this in the [3.3.0-beta.1](https://github.com/seriousm4x/UpSnap/releases/tag/3.3.0-beta.1).

Devices without groups will be on top followed by the groups sorted alphabetically. New groups can be created and deleted on the settings page and devices can be assigned to groups in the device settings.

Could you both @artfer @gaby try it out and tell me if its easy to use or something breaks?

**Comment by artfer** [comment]
This looks great and is easy and intuitive to use. Nice!

**Comment by gaby** [comment]
Looks great! Is there a way to add them to a group via the API?

**Comment by artfer** [comment]
Should've checked for bugs before replying... When I press to turn on a machine everything gets duplicated and the same happens to turn on, but the state remains on the old cards. It seems it duplicates the cards for each unique state of the machine.

**Comment by seriousm4x** [comment]
> When I press to turn on a machine everything gets duplicated and the same happens to turn on, but the state remains on the old cards. It seems it duplicates the cards for each unique state of the machine.

Oh, I will take a look at that.

> `[ERROR] 2023/06/30 15:15:30 handlers.go:46: `

Are there no errors after that? You could try to shutdown from inside the container `docker exec -it upsnap sh` and then your shutdown cmd. Feel free to open a seperate issue if that doesn't help finding the issue.

**Comment by artfer** [comment]
> Are there no errors after that? 

Strangely that is all it outputs. However, it was my fault. As I updated the docker image I lost the ssh key I was using before. Created a new one and it still works. Sorry.

I'll leave you with a screenshot of when I turned on one of my machines,
![image](https://github.com/seriousm4x/UpSnap/assets/36967265/2ce73040-8881-45d0-9bb7-4826f80cb583)
There are only 2 computers here but they appear many times. It replicates from left to right, and when the row is full, it uses another row.
